src/qt/guiutil.cpp

```diff
@@ -857,10 +857,10 @@ void restoreWindowGeometry(const QString& strSetting, const QSize& defaultSize,
 
     parent->resize(size);
     parent->move(pos);
-
-    if ((!pos.x() && !pos.y()) || (QApplication::desktop()->screenNumber(parent) == -1))
-    {
-        QRect screen = QApplication::desktop()->screenGeometry();
+    
+    int screen_number = QApplication::desktop()->screenNumber(parent);
+    QRect screen = QApplication::desktop()->availableGeometry(screen_number);
+    if ((!pos.x() && !pos.y()) || screen_number == -1 || !screen.contains(parent->frameGeometry())) {
```

Comment:
If I understand correctly, `!pos.x() && !pos.y()` should be replaced with `!settings.contains(strSettings + "Pos")`.

Reply:
Safer to check after converting it to a point, just to make sure conversion worked properly?

Reply:
But what if the window position is `QPoint(0, 0)`?